legend-engine-xt-changetoken-pure/src/main/resources/core_pure_changetoken/cast_generation.pure

```diff
@@ -454,6 +467,110 @@ function <<access.private>> meta::pure::changetoken::cast_generation::_handleAdd
   );
 }
 
+function <<access.private>> meta::pure::changetoken::cast_generation::_inlineJsonValue(
+key: String[1],
+json:JSONElement[1]
+) : Code[1]
+{
+  simpleEntry()->j_new([j_string($key),
+    $json->match([
+      b:JSONBoolean[1] | j_boolean($b.value),
+      s:JSONString[1] | j_string($s.value),
+      n:JSONNumber[1] | $n.value->match([i:Integer[1] | j_int($i), f:Float[1] | j_double($f)]),
+      o:JSONObject[1] | javaStream()->j_invoke('of',
+      $o.keyValuePairs->map(kv | _inlineJsonValue($kv.key.value, $kv.value)),
+      javaVoid())->j_invoke('collect', [javaCollectors()->j_invoke('toMap', [j_methodReference(AbstractMap_SimpleEntry(), 'getKey', javaFunctionType([], javaObject())), j_methodReference(AbstractMap_SimpleEntry(), 'getValue', javaFunctionType([], javaObject()))], javaVoid())], javaVoid()),
+      n:JSONNull[1] | j_null()
+    ])
+  ])
+}
+
+function <<access.private>> meta::pure::changetoken::cast_generation::_isComplex(
+value: Any[1]
+) : Boolean[1]
+{
+  if($value->instanceOf(String) && $value->cast(@String)->startsWith('{') && $value->cast(@String)->endsWith('}'),
+     | true,
+     | false
+  );
+}
+
+function <<access.private>> meta::pure::changetoken::cast_generation::_isPrimitive(
+type: String[1]
+) : Boolean[1]
+{
+   if($type->equalIgnoreCase('Boolean[1]'),
+      | true,
+      |
+   if($type->equalIgnoreCase('Float[1]') || $type->equalIgnoreCase('Double[1]'),
+      | true,
+      |
+   if($type->equalIgnoreCase('Integer[1]'),
+      | true,
+      |
+   if($type->equalIgnoreCase('String[1]'),
+      | true,
+      | false
```

Comment:
We should support all and only the primitive Pure types here. Double, for example, is not a Pure primitive type. Date, DateTime, StrictDate, StrictTime, Decimal are some that are missing here.

Also, I would question the use of equalIgnoreCase here.